### ISSUE TITLE
wrong path normalize

### ISSUE THREAD

**Issue by f0re1gnKey** [oracle/graal#6417]
[INFO] Executing: D:\Portable Program Files\graalvm\bin\native-image.cmd @target\tmp\native-image-13255282300971752693.args
Exception in thread "main" java.nio.file.InvalidPathException: Illegal char < > at index 3: D:\ My Data\ IDE????\ xxxxxxxxxxx\ xxxxxxxxxxxx\ IDEA\ target\ classes
	at java.base@17.0.6/sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:182)
	at java.base@17.0.6/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:153)
	at java.base@17.0.6/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
	at java.base@17.0.6/sun.nio.fs.WindowsPath.parse(WindowsPath.java:92)
	at java.base@17.0.6/sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:232)
	at java.base@17.0.6/java.nio.file.Path.of(Path.java:147)
	at java.base@17.0.6/java.nio.file.Paths.get(Paths.java:69)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.util.ClasspathUtils.stringToClasspath(ClasspathUtils.java:52)
	at org.graalvm.nativeimage.driver/com.oracle.svm.driver.NativeImage.addCustomImageClasspath(NativeImage.java:1582)
	at org.graalvm.nativeimage.driver/com.oracle.svm.driver.DefaultOptionHandler.processClasspathArgs(DefaultOptionHandler.java:417)
	at org.graalvm.nativeimage.driver/com.oracle.svm.driver.DefaultOptionHandler.consume(DefaultOptionHandler.java:69)
	at org.graalvm.nativeimage.driver/com.oracle.svm.driver.NativeImage$NativeImageArgsProcessor.apply(NativeImage.java:1512)
	at org.graalvm.nativeimage.driver/com.oracle.svm.driver.DefaultOptionHandler.consume(DefaultOptionHandler.java:204)
	at org.graalvm.nativeimage.driver/com.oracle.svm.driver.NativeImage$NativeImageArgsProcessor.apply(NativeImage.java:1512)
	at org.graalvm.nativeimage.driver/com.oracle.svm.driver.NativeImage.processNativeImageArgs(NativeImage.java:1770)
	at org.graalvm.nativeimage.driver/com.oracle.svm.driver.NativeImage.completeImageBuild(NativeImage.java:959)
	at org.graalvm.nativeimage.driver/com.oracle.svm.driver.NativeImage.build(NativeImage.java:1417)
	at org.graalvm.nativeimage.driver/com.oracle.svm.driver.NativeImage.performBuild(NativeImage.java:1387)
	at org.graalvm.nativeimage.driver/com.oracle.svm.driver.NativeImage.main(NativeImage.java:1374)

**Comment by fniephaus** [comment]
Thanks for opening the ticket. What version of GraalVM are you using? Could you please try building with a [dev build](https://github.com/graalvm/graalvm-ce-dev-builds/releases)? We've fixed an issue similar to this one some time ago.

**Comment by tlf30** [comment]
I have the same issue, running version `graalvm-ce-java17-windows-amd64-22.3.2`
I can confirm that GraalVM CE 23.1.0-dev-20230505_0633 fixes the issue.

**Comment by KaeYan93** [comment]
Hi, I encountered the same issue with 22.3.3. Can i confirm if it is not resolved for version 22.3.3?

**Comment by fniephaus** [comment]
Could you try again with GraalVM for JDK 17 or later?

**Comment by KaeYan93** [comment]
Thanks @fniephaus for reply, can i confirm if GraalVM for JDK 17.0.8 can fix this? Because if i were to use 17.0.9, I would have to upgrade my Windows visual studio version.

**Comment by KaeYan93** [comment]
Hi @fniephaus , thanks for the reply. I tried 17.0.8 and it works. Can i confirm if the below docker image is the 17.0.8? 
https://github.com/graalvm/container/pkgs/container/graalvm-ce/112515416?tag=ol9-java17

Currently when i `java -version`, this is the print out:
```
java version "17.0.8" 2023-07-18 LTS
Java(TM) SE Runtime Environment (build 17.0.8+9-LTS-211)
Java HotSpot(TM) 64-Bit Server VM (build 17.0.8+9-LTS-211, mixed mode, sharing)
```
